model.py

```diff
@@ -8812,42 +8812,95 @@ class ExternalIntegration(Base):
     to a third-party API.
     """
 
-    # Utilities
+    # Possible types of ExternalIntegrations.
+    #
+    # These integrations are associated with external services such as
+    # Google Enterprise which authenticate library administrators.
+    ADMIN_AUTH_TYPE = 'admin_auth'
+
+    # These integrations are associated with external services such as
+    # SIP2 which authenticate library patrons. Other constants related
+    # to this are defined in the circulation manager.
+    PATRON_AUTH_TYPE = 'patron_auth'
+
+    # These integrations are associated with external services such
+    # as Overdrive which provide access to books.
+    LICENSE_TYPE = 'licenses'
+
+    # These integrations are associated with external services such as
+    # the metadata wrangler, which provide information about books,
+    # but not the books themselves.
+    METADATA_TYPE = 'metadata'
+
+    # These integrations are associated with external services such as
+    # Google Analytics, which receive analytics events.
+    ANALYTICS_TYPE = 'analytics'
+       
+    # These integrations are associated with the Library Simplified
+    # applications themselves, as well as external services such as
+    # CDNs and Elasticsearch which are directly used by the
+    # applications.
+    SYSTEM_TYPE = 'system'
```

Comment:
As opposed to having single protocols for CDN or S3, and different integration types for the different uses of those.